### ISSUE TITLE
generator terminates one line before eof

### ISSUE THREAD

**Issue by kaiweiang** [knights-lab/shi7#39]
When trying a function such as this, on a file of known 500 sequences, I always get 499 back (499 headers, seqs, and quality strings):

```

def check_sequence_name(path):
    ids = []
    print(path)
    vari = 1
    with open(path) as path_inf:
        fastq_gen = read_fastq(path_inf)
        for gen in fastq_gen:
            print(vari,"...",gen[0],"\n",gen[1],"\n",gen[2],"\n")
            vari += 1
            ids.append(gen[0][-9]) # we can't always assume it's this position
    return ids, len(ids) #why the output has size of 499 (Ben: skipping last line!)
```

**Comment by bhillmann** [comment]
I updated the read_fastq code and created a test file. Bug should be fixed in latest commit.


```
def read_fastq(fh):
    qual_flag = False
    data = ''
    title = next(fh)[1:].strip()
    qualities = ''
    for line in fh:
        line = line.strip()
        if len(data) == len(qualities) and qual_flag:
            yield title, data, qualities
            qual_flag = False
            title = line[1:]
            data = ''
            qualities = ''
        elif qual_flag:
            qualities += line
        elif len(line) == 1 and line[0] == '+':
            qual_flag = True
        else:
            data += line
    yield title, data, qualities
```

**Comment by bhillmann** [comment]
This is resolved in 444ba491a68ac2e764e8fd3292ef1bdeb4a2fb20

**Comment by bhillmann** [comment]
This is resolved in 444ba491a68ac2e764e8fd3292ef1bdeb4a2fb20
